packages/frontend/src/creation-form/components/approve-rewards.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+    type KPITokenCreationFormProps,
+    type NamespacedTranslateFunction,
+    TxType,
+} from "@carrot-kpi/react";
+import { Button } from "@carrot-kpi/ui";
+import { type ReactElement, useCallback, useState, useEffect } from "react";
+import { type Address, useContractReads, useAccount } from "wagmi";
+import { zeroAddress, type TransactionReceipt } from "viem";
+import type { Reward } from "../types";
+import { ERC20_ABI } from "@carrot-kpi/sdk";
+import { ApproveReward } from "./approve-button";
+import { dateToUnixTimestamp } from "../../utils/dates";
+
+interface ApproveRewardsProps {
+    t: NamespacedTranslateFunction;
+    rewards?: Reward[];
+    owner?: Address;
+    spender: Address;
+    onApprove: () => void;
+    onTx: KPITokenCreationFormProps<object>["onTx"];
+}
+
+export const ApproveRewards = ({
+    t,
+    rewards,
+    spender,
+    onApprove,
+    onTx,
+}: ApproveRewardsProps): ReactElement => {
+    const { address: connectedAddress } = useAccount();
+
+    const [currentIndex, setCurrentIndex] = useState(0);
+    const [toApprove, setToApprove] = useState<Reward[]>(rewards || []);
+    const [allApproved, setAllApproved] = useState(false);
+    const currentlyApprovingReward: Reward | undefined =
+        toApprove[currentIndex];
+
+    const { data: allowances, isLoading: loadingAllowances } = useContractReads(
+        {
+            contracts:
+                connectedAddress &&
+                rewards?.map((reward) => {
+                    return {
+                        address: reward.address,
+                        abi: ERC20_ABI,
+                        functionName: "allowance",
+                        args: [connectedAddress, spender],
+                    };
+                }),
```

Comment:
Wondering if it makes sense to add `enabled: !!connectedAddress && !!rewards` here?

Reply:
You're right, addressed in cb4183d147f13276e725c1711381162a0a69da09